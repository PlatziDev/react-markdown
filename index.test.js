const React = require('react');
const renderer = require('react-test-renderer');
const Markdown = require('./index.js');

const content = `![Platzi Logo](https://static.platzi.com/static/images/logos/platzi@2x.png)
*[HTML]: Hyper Text Markup Language
*[W3C]:  World Wide Web Consortium
# Heading 1
Lorem **ipsum** dolor _sit_ ++amet++ :D.
The HTML ==specification== is maintained by the W3C.
- \`item 1\`
- item 2
- item 3
Term 1
  ~ Definition 1
Term 2
  ~ Definition 2a
  ~ Definition 2b
\`\`\`js
function greeting() {
	return 'hello world!';
}
\`\`\`
@[youtube](ajLJOhf-WdA)`;


test('Markdown component', () => {
  expect(
    renderer.create(
      React.createElement(Markdown, { content })
    ).toJSON()
  ).toMatchSnapshot();
});
