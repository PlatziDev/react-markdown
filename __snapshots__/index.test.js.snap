// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown component 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<figure><img src=\\"https://static.platzi.com/static/images/logos/platzi@2x.png\\" alt=\\"Platzi Logo\\" /></figure>
    <h1>Heading 1</h1>
    <p>Lorem <strong>ipsum</strong> dolor <em>sit</em> <ins>amet</ins> ðŸ˜„.<br />
    The <abbr title=\\"Hyper Text Markup Language\\">HTML</abbr> <mark>specification</mark> is maintained by the <abbr title=\\"World Wide Web Consortium\\">W3C</abbr>.</p>
    <ul>
    <li><code>item 1</code></li>
    <li>item 2</li>
    <li>item 3<br />
    Term 1<br />
    ~ Definition 1<br />
    Term 2<br />
    ~ Definition 2a<br />
    ~ Definition 2b</li>
    </ul>
    <pre><code class=\\"language-js\\">function greeting() {
    	return 'hello world!';
    }
    </code></pre>
    <p><div class=\\"embed-responsive embed-responsive-16by9\\"><iframe class=\\"embed-responsive-item\\" id=\\"youtubeplayer\\" type=\\"text/html\\" width=\\"640\\" height=\\"390\\" src=\\"//www.youtube.com/embed/ajLJOhf-WdA\\" frameborder=\\"0\\" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div></p>
    ",
    }
  }
  id="content"
/>
`;
